xformers/components/multi_head_dispatch.py

```diff
@@ -79,33 +79,36 @@ def __init__(
         dim_key, dim_value = map(lambda x: x if x else dim_model, (dim_key, dim_value))
 
         self.num_heads = num_heads
-        self.dim_k = dim_key // num_heads
-        self.dim_value = dim_value
+        self.dim_key_head = dim_key // num_heads
+        self.dim_value_head = dim_value // num_heads
         self.dim_model = dim_model
         self.attention = attention
 
         # key, query, value projections for all heads
         # critical options are
         # - are we sharing weights ?
-        # - are we adding biases, and if yes are they shared ?
+        # - are we adding biases ?
```

Comment:
I'm not sure of the amount of nuances to support here, for instance one bias setting per input ? could be a little overkill, but easy to add if need be